scripts/transformations.py

```diff
@@ -20,130 +20,137 @@
-    new_transformMatrix[0:0+matr_rotate.shape[0], 0:0+matr_rotate.shape[1]] = matr_rotate
-    new_transformMatrix[0:0+matr_translate.shape[0], 3:3+matr_translate.shape[1]] = matr_translate
-    new_transformMatrix[new_transformMatrix.shape[0]-1,new_transformMatrix.shape[1]-1] = 1
-
-    ## Return result
-    return new_transformMatrix
-
-  def convertPath2FixedFrame(self, path_body, frame_body, frame_fixed=np.identity((4))):
-    """
-    Function for mapping a Path (type: List) onto a Fixed Frame (np. Homogenous Matrix)
-    :param path_body: Python List of Transformation Matrices for each point along a path
-    :param frame_body: Transformation matrix for the body
-    :param frame_fixed: Defaults to Identity Matrix (or robot's fixed frame)
-    """
-
-    ## Find Arb Frame Defined By Fixed Frame
-    frame_b2f = np.matmul(frame_body, frame_fixed)  #Pre-Multiply bc wrt fixed frame
-
-    ## Convert Path Frames to be defined by Fixed Frame
-    path_fixed = []
-    for point in path_body:
-      path_fixed.append(np.matmul(frame_b2f, point))
-
-    return path_fixed
-
-  def convertPath2RobotPose(self, path):
-    """
-    Convert Path (transforms) to List of Robot Poses (xyz,quants)
-    Function based upon math in...
-    https://en.wikipedia.org/wiki/Rotation_matrix#Quaternion
-    :param path: List of Homogeneous Transformations
-    :return: List of Robot Poses
-    """
-
-    ## Imports
-    import numpy as np
-
-    ## Loop through Path Transforms and convert to Poses
-    path_poses = []
-    for point in path:
-
-      # Location Vector
-      x,y,z = point[:-1,3]
-      x = np.asscalar(x)
-      y = np.asscalar(y)
-      z = np.asscalar(z)
-
-      # Quant Calculation Support Variables
-      # Only find trace for the rotational matrix.
-      t = np.trace(point) - point[3,3]
-      r = np.sqrt(1+t)
-
-      # Primary Diagonal Elements
-      Qxx = point[0,0]
-      Qyy = point[1,1]
-      Qzz = point[2,2]
-
-      ##TODO: Fix (invalid value warning) so copysign quant calculation handles imaginary values (or only uses real?)
-      #print(Qxx, Qyy, Qzz)
-      #print(np.real(np.sqrt(1 - Qxx - Qyy + Qzz)))
-
-      # Quant Calculation
-      # Using np.lib.scimath.sqrt in lieu of np.sqrt as it returns imaginary components vs NaN
-      # np.absolute used to get magnitude of combined real + imaginary components 
-      qx = np.copysign(np.absolute(0.5 * np.lib.scimath.sqrt(1 + Qxx - Qyy - Qzz)), point[2,1]-point[1,2])
-      qy = np.copysign(np.absolute(0.5 * np.lib.scimath.sqrt(1 - Qxx + Qyy - Qzz)), point[0,2]-point[2,0])
-      qz = np.copysign(np.absolute(0.5 * np.lib.scimath.sqrt(1 - Qxx - Qyy + Qzz)), point[1,0]-point[0,1])
-      qw = 0.5*r
-
-      if np.isnan(qx): qx = 0
-      if np.isnan(qy): qy = 0
-      if np.isnan(qz): qz = 0
-
-      path_poses.append( [x, y, z, qx,qy,qz, qw] )
-
-    return path_poses
-
-##########################
-
+from typing import Union
+from typing import List
```

Comment:
These imports are not used. Further they are very close to existing reserved names of `list`.
Why do these need included?

Reply:
They are for the type assistance / type hints under after function - `List` is intentionally close to `list`. Though I don't know *exactly* how the type assists work this is the proper way to signify a list. (`Union` signifies that a function can accept multiple input type for one argument).

They aren't and shouldn't be used in the code itself per se but are needed for the type assistance. [See here](https://mypy.readthedocs.io/en/stable/python2.html)

The short of it is that they should stay but I'm glad you pointed out this potential confusion!

Reply:
Not a huge fan of including these imports just for the type assistance as this is a not a standard install Python module and would introduce a bug that users would consistently get when setting up environment.

We could including `typing` in the `ros-setup` bash script to ensure inclusion. And also including `typing` in the package`__init.py__` .

So, is this important enough you want to keep it?

Reply:
Resolved by adding requirements.txt